syntax = "proto3";

option go_package = "go/predict";

service PredictService {
  rpc MakePredict (MakePredictRequest) returns (MakePredictResponse);
  rpc GetPredict (GetPredictRequest) returns (GetPredictResponse);
}

message TimeSeriesItem {
  int64 ts = 1;
  double value = 2;
}

message MakePredictRequest {
  string username = 1;
  string name = 2;
  string unit = 3;
  int32 period = 4;
  bytes series = 5;
}

message MakePredictResponse {
  repeated TimeSeriesItem items = 1;
}

message GetPredictRequest {
  string username = 1;
  string name = 2;
}

message GetPredictResponse {
  string unit = 1;
  int64 delimiter = 2;
  repeated TimeSeriesItem items = 3;
}
